posthog/api/event.py

```diff
@@ -163,3 +167,71 @@ def values(self, request: request.Request) -> response.Response:
         """.format(where), params)
 
         return response.Response([{'name': value.value} for value in values])
+
+    @action(methods=['GET'], detail=False)
+    def sessions(self, request: request.Request) -> response.Response:
+        events = self.get_queryset().filter(**request_to_date_query(request.GET)) # type: ignore
```

Comment:
I think request_to_date should be in the _filter_request above rather than here. It's slightly annoying to wrangle that in but it means we can filter the entire endpoint using those dates. Also what's the type error you got here?

Reply:
Ok, will try to merge the filtering into the filter_request.

Type error is QueryDict vs Dict[str, Any]. The request_to_date_query worked in other places because the response wasn't explicitly defined in the params but since it is here, the params are a QueryDict. I can explicitly convert

Reply:
Yep worth converting explicitly

Reply:
One gotcha with trying to insert `request_to_date_query` into _filter_request is that `request_to_date_query` defaults to a range even if there aren't explicitly passed params for date_from and date_to which is how trends page makes calls. I could drop it into the get_queryset when the action is 'sessions' otherwise putting it in _filter_request will affect all other `/event/...` calls

Reply:
Good point, hadn't thought about that. Let's leave as is for now then.